provider/execution/execution.go

```diff
@@ -359,21 +359,34 @@ func (e TableExecutor) resolveColumns(ctx context.Context, meta schema.ClientMet
 		}
 		meta.Logger().Trace("setting column value", "column", c.Name, "value", v, "table", e.Table.Name)
 		if err := resource.Set(c.Name, v); err != nil {
-			return FromError(err, WithResource(e.ResourceName), WithType(diag.INTERNAL),
-				WithSummary("failed to set resource value for column %s@%s", e.Table.Name, c.Name))
+			return FromError(err, diag.WithResourceName(e.ResourceName), diag.WithType(diag.INTERNAL),
+				diag.WithSummary("failed to set resource value for column %s@%s", e.Table.Name, c.Name))
 		}
 	}
 	return nil
 }
 
 // handleResolveError handles errors returned by user defined functions, using the ErrorClassifiers if defined.
-func (e TableExecutor) handleResolveError(meta schema.ClientMeta, err error, opts ...Option) diag.Diagnostics {
+func (e TableExecutor) handleResolveError(meta schema.ClientMeta, r *schema.Resource, err error, opts ...diag.BaseErrorOption) diag.Diagnostics {
+	errWithDiag := FromError(err, append(opts,
+		diag.WithResourceName(e.ResourceName),
+		WithResource(r),
+		diag.WithSeverity(diag.ERROR),
+		diag.WithType(diag.RESOLVING),
+		diag.WithSummary("failed to resolve table %q", e.Table.Name),
+	)...)
+
+	classifiedDiags := make(diag.Diagnostics, 0, len(errWithDiag))
 	for _, c := range e.classifiers {
-		if diags := c(meta, e.ResourceName, err); diags != nil {
-			return diags
+		for _, d := range errWithDiag {
+			if diags := c(meta, e.ResourceName, d); diags != nil {
+				classifiedDiags = classifiedDiags.Add(diags)
+			}
 		}
 	}
```

Comment:
would document this part as it not 100% obvious. Basically  error can be diag.Diagnostics so we in FromError return it as is and we want to iterate over them.

Reply:
Documented